src/editor.js

```diff
@@ -2874,34 +2875,68 @@ config.defineOptions(Editor.prototype, "editor", {
 
-            var keyboardFocusClassName = "ace_keyboard-focus";
+            var gutterKeyboardHandler;
 
             // Prevent focus to be captured when tabbing through the page. When focus is set to the content div, 
             // press Enter key to give focus to Ace and press Esc to again allow to tab through the page.
             if (value){
+                this.keyboardFocusClassName = "ace_keyboard-focus";
+
                 this.textInput.getElement().setAttribute("tabindex", -1);
                 this.renderer.content.setAttribute("tabindex", 0);
-                this.renderer.content.classList.add(keyboardFocusClassName);
+                this.renderer.content.setAttribute("role", "group");
+                this.renderer.content.setAttribute("aria-roledescription", "editor");
+                this.renderer.content.classList.add(this.keyboardFocusClassName);
                 this.renderer.content.setAttribute("aria-label",
-                    "Editor, press Enter key to start editing, press Escape key to exit"
+                    "Editor, press Enter to start editing, press Escape to exit"
                 );
 
                 this.renderer.content.addEventListener("keyup", focusOnEnterKeyup.bind(this));
                 this.commands.addCommand(blurCommand);
+
+                this.renderer.$gutter.setAttribute("tabindex", 0);
+                this.renderer.$gutter.setAttribute("aria-hidden", false);
+                this.renderer.content.setAttribute("role", "group");
+                this.renderer.content.setAttribute("aria-roledescription", "gutter");
+                this.renderer.$gutter.setAttribute("aria-label",
+                    "Gutter, press Enter to interact with controls using arrow keys, press Escape to exit"
+                );
+                this.renderer.$gutter.classList.add(this.keyboardFocusClassName);
+
+                if (!gutterKeyboardHandler)
+                    gutterKeyboardHandler = new GutterKeyboardHandler(this);
+
+                gutterKeyboardHandler.addListener();
             } else {
                 this.textInput.getElement().setAttribute("tabindex", 0);
                 this.renderer.content.setAttribute("tabindex", -1);
-                this.renderer.content.classList.remove(keyboardFocusClassName);
+                this.renderer.content.setAttribute("role", "");
+                this.renderer.content.setAttribute("aria-roledescription", "");
+                this.renderer.content.classList.remove(this.keyboardFocusClassName);
                 this.renderer.content.setAttribute("aria-label", "");
             
                 this.renderer.content.removeEventListener("keyup", focusOnEnterKeyup.bind(this));
                 this.commands.removeCommand(blurCommand);
+
+                this.renderer.$gutter.setAttribute("tabindex", -1);
+                this.renderer.$gutter.setAttribute("aria-hidden", true);
+                this.renderer.content.setAttribute("role", "");
```

Comment:
What is a best approach for removing the aria attributes, is it set them to empty value or remove them completely. I guess I followed a different approach and asking to be consistent.

Reply:
I don't think there are differences between how assistive tech interpret it but it probably makes more sense to just remove the attributes. I'll update the PR.